src/Nethermind/Nethermind.Blockchain/Synchronization/SyncServer.cs

```diff
@@ -128,81 +145,91 @@ public void AddNewBlock(Block block, Node nodeWhoSentTheBlock)
             }
 
             if ((block.TotalDifficulty ?? 0) < _blockTree.BestSuggestedHeader.TotalDifficulty) return;
+            block.Header.TotalDifficulty = null; // important line - read the long comment above
```

Comment:
Do we want to do this always, not only on potential issue?

Reply:
I analysed and tested various approached.
1) I tried approach where BlockValidator was testing it - it caused trouble
2) I tried the case where BlockTree was doign that internally - it was better but then the logic internally would be complex or would require additional parameters or methods exposed by the BlockTree. I tried changing BlockTree but that felt a bit too invasive (I tried two or three approches with various level of checks and resiliency).
3) in the end I realized that the only place where we do not trust TotalDifficulty is in AddNewBlock.

Now, for the last year or so I was planning to move all the validation inside the BlockTree but this is non-trivial. There are sync cases where validation has to happen outside of the tree at the moment. I think it can be refactored properly but this is 6 - 8 hours work piece.